### DISCUSSION TITLE
How to flash the bin?

### DISCUSSION THREAD

**Discussion by jmvaz** [gysmo38/mitsubishi2MQTT#204] [Q&A]
Can anyone describe me the steps how to flash the bin on the esp? Esphome flasher seems not to work with the current released bin.

Thanks

**Comment by jmvaz** [comment]
Anyone?

**Reply by lesleypersyn** [reply]
Please try another bin first to check what the problem exactly is... and please describe 'not to work'.
Make sure you understand flashing and clearing ESP's....

**Comment by Noori1972** [comment]
Flashing firmware onto an ESP8266 module typically involves using a tool like esptool or the Arduino IDE. Since you mentioned that the ESP Home Flasher is not working with the current released bin, you might need to use esptool, which is a command-line tool specifically designed for flashing firmware onto ESP8266 and ESP32 devices. Here's a general guide on how to use esptool to flash a firmware binary onto an ESP8266:

Download esptool: If you don't have esptool installed, you can download and install it by following the instructions in the official GitHub repository: https://github.com/espressif/esptool

Download the Firmware Binary: Make sure you have the firmware binary that you want to flash onto the ESP8266. This should be in the .bin format.

Connect ESP8266: Connect your ESP8266 module to your computer using a USB-to-Serial adapter or any other appropriate method. Make sure you know which serial port your ESP8266 is connected to.

Put ESP8266 into Flash Mode: To put the ESP8266 into flashing mode, you usually need to pull the GPIO0 pin low while resetting the board. This varies depending on your specific board and setup. Consult your board's documentation for details.

Open a Terminal/Command Prompt: Open a terminal or command prompt on your computer.

Navigate to esptool Directory: Use the cd command to navigate to the directory where you have the esptool.py script. This is the script you downloaded in step 1.

Run Flashing Command: Run the following command to flash the firmware binary onto the ESP8266:

esptool.py --port /dev/ttyUSB0 write_flash -fm dio -fs 32m 0x00000 path_to_firmware.bin

Replace /dev/ttyUSB0 with the appropriate serial port for your setup.
Replace path_to_firmware.bin with the actual path to your firmware binary.
The -fm dio flag specifies the flash mode, and -fs 32m specifies the flash size. These values might differ based on your specific ESP8266 module.

Monitor the Flashing Process: The flashing process will start, and you'll see progress information in the terminal/command prompt. It will erase the flash and write the new firmware.

Verify and Test: Once the flashing process is complete, you can reset your ESP8266. It should now boot into the new firmware.

Remember that flashing firmware carries a risk of damaging your device if not done correctly. Make sure to double-check your connections and consult the documentation for your specific ESP8266 module to ensure you're using the right pins and settings.

Additionally, consider exploring community forums or resources related to your specific ESP8266 module, as there might be additional insights or recommendations based on your module's characteristics.
